bokehjs/src/coffee/model.coffee

```diff
@@ -1,13 +1,13 @@
 _ = require "underscore"
 HasProps = require "./core/has_props"
+p = require "./core/properties"
 
 class Model extends HasProps
   type: "Model"
 
-  defaults: ->
-    return _.extend {}, super(), {
-      tags: []
-      name: null
-    }
+  props: () -> _.extend super(), {
```

Comment:
You have to decide whether you use `{}` as the first argument or not, and do it consistently. The reason we use `{}` is to make sure `_.extend` doesn't modify `super()` in-place. As long as you can assure that `props()` always returns a fresh instance of an object, then `{}` is unnecessary. However, if at any point `props()` returns a reference to, lets say, a class member, then it will modify the global state and lead to a subtle bug.